packages/react/src/auto/AutoForm.ts

```diff
@@ -73,6 +74,56 @@ export const useFormFields = (
           } as const)
       )
     );
+
     return [...includedObjectFields, ...includedRootLevelFields];
   }, [metadata, options]);
 };
+
+/**
+ * Internal React hook for sharing logic between different `AutoForm` components.
+ * @internal
+ */
+export const useAutoForm = <
+  GivenOptions extends OptionsType,
+  SchemaT,
+  ActionFunc extends ActionFunction<GivenOptions, any, any, SchemaT, any>,
+  Options extends ActionFunc["optionsType"]
+>(
+  props: AutoFormProps<GivenOptions, SchemaT, ActionFunc, Options>
+) => {
+  const { action, record, findBy } = props;
+
+  const { metadata, fetching: fetchingMetadata, error: metadataError } = useActionMetadata(props.action);
+
+  // filter down the fields to render only what we want to render for this form
+  const fields = useFormFields(metadata, props);
+  const operatesWithRecordId = !!metadata?.action.operatesWithRecordIdentity;
+
+  // setup the form state for the action
+  const {
+    submit,
+    error: formError,
+    formState: { isSubmitSuccessful, isLoading },
+    originalFormMethods,
+  } = useActionForm(action, {
+    defaultValues: {
+      [action.modelApiIdentifier]: record,
+      id: "0", // The ID value will be replaced when sending the form to use the record found by `findBy`
```

Comment:
The `id` is marked as a required field in the validator, so I have to add a dummy id value to it.

Inside the `useActionForm`, we will replace the value from the record found when we submit the form:
https://github.com/gadget-inc/js-clients/blob/7acce09540392a06100085a2054813960dd95822/packages/react/src/useActionForm.ts#L186-L188